android/src/main/java/com/adyenreactnativesdk/ui/LoadingDialogFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.adyenreactnativesdk.ui
+
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.DialogFragment
+import com.adyenreactnativesdk.R
+
+interface Cancelable {
+    fun canceled()
+}
+
+class LoadingDialogFragment(private val cancelable: Cancelable) : DialogFragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        isCancelable = false
+        return inflater.inflate(R.layout.loading, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        requireDialog().window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        val button = view.findViewById<Button>(R.id.loading_button)
+        button.setOnClickListener {
+            cancelable.canceled()
+        }
+    }
+
+    companion object {
+        fun newInstance(cancelable: Cancelable): LoadingDialogFragment {
+            return LoadingDialogFragment(cancelable)
```

Comment:
instead of passing the parameter in the constructor you should put it in arguments so your `newInstance()` 
```kotlin
val fragment = LoadingDialogFragment()
val args = Bundle().apply {
    putParcelable(KEY_CANCELLABLE, cancelable)
}
fragment.arguments = args
return fragment
}

but then you will have to make your `Cancelable` implementations `Parcelable`. you can use `@Parcelize` annotation for that. (you can take a look at this pr to see how to do that https://github.com/Adyen/adyen-android/pull/834)